[Reviewer: Matt] - Replaced old LVC command with SNMP command

TygerBurning opened the pull request:
This update now passes the `make resync_test` within Chronos. It reports the same value, and flags are used to make the SNMP command only return the value.

One thing Rob mentioned is that we use the `-c` flag here, which has previously been confined to a particular file? Users would also need to update this file to customise. Worth a follow up/documenting?

mirw commented on debian/chronos.init.d:
```diff
@@ -133,12 +133,12 @@ do_wait_sync() {
   do
     # Retrieve the statistics.
-    nodes=`/usr/share/clearwater/chronos/bin/cw_stat chronos chronos_scale_nodes_to_query`
+    nodes=`snmpget -Oqv -M. -v2c -c clearwater sprout-1.ajw3.cw-ngv.com .1.2.826.0.1.1578918.9.10.1`
```
`sprout-1.ajw3.cw-ngv.com` isn't going to be right.

mirw replied:
Is `-c clearwater` OK?  I think we should add a separate internal-only community for this.

mirw replied:
What if `clearwater-snmpd` isn't installed?

mirw replied:
What if `snmp` isn't installed?

graemerobertson replied:
This line generally 'looks good' though

mirw commented on debian/chronos.init.d:
```diff
@@ -147,7 +147,7 @@ do_wait_sync() {
-    if [ "$nodes" = 'Unknown statistic "chronos_scale_nodes_to_query"' ]
+    if [ "$nodes" = 'Invalid OID: ".1.2.826.0.1.1578918.9.10.1"' ]
     then
       echo "SNMP statistics are not supported on this box. Please install the required packages and retry"
```
This is a bit worrying - surely it's a bit late now?  I think we should either mandate installing snmp/snmpd or drop out earlier.

mirw commented:
Passing back - I think you and Rob were discussing.  Either way, I've finished my review.

TygerBurning commented:
Modified to localhost version. Also added snmp and clearwater-snmpd to the Depends section in debian/control, which allows us to remove the case that SNMP is not installed.

eleanor-merry commented on debian/control:
```diff
@@ -8,7 +8,7 @@ Homepage: http://github.com/Metaswitch/chronos
 Package: chronos
 Architecture: any
-Depends: clearwater-infrastructure, libboost-program-options1.54.0, libboost-regex1.54.0, libboost-filesystem1.54.0, libevent-2.0-5, libevent-pthreads-2.0-5, libc-ares-dev, clearwater-snmp-handler-chronos, clearwater-log-cleanup, libsnmp30
+Depends: clearwater-infrastructure, libboost-program-options1.54.0, libboost-regex1.54.0, libboost-filesystem1.54.0, libevent-2.0-5, libevent-pthreads-2.0-5, libc-ares-dev, clearwater-snmp-handler-chronos, clearwater-log-cleanup, libsnmp30, clearwater-snmpd, snmp
```
I think you can remove the clearwater-snmp-handler-chronos dependency as well (but not as part of this PR)

eleanor-merry commented on debian/chronos.init.d:
```diff
@@ -133,26 +133,20 @@ do_wait_sync() {
+    nodes=`snmpget -Oqv -M. -v2c -c clearwater localhost .1.2.826.0.1.1578918.9.10.1`
 
     # If the nodes left to query is 0 or unset, we're finished
-    if [ "$nodes" = "0" ] || [ "$nodes" = "No value returned" ]
```
After discussion with @rkd-msw I think we do still need to check for 'No Such Instance currently exists at this OID', and exit with a warning if we hit this, or better yet, exit if we hit this 3 times in a row. @TygerBurning - would you be able to implement this?

TygerBurning commented:
Updated with bash to drop out if our query fails 3 times

eleanor-merry commented on debian/chronos.init.d:
```diff
@@ -131,26 +131,31 @@ do_scale_operation()
 
-    # If the statistic is unknown then the SNMP handler isn't installed. Give up
-    if [ "$nodes" = 'Unknown statistic "chronos_scale_nodes_to_query"' ]
+    # If we have 3 failures at getting the nodes, give up
```
This should be 3 failures in a row - I think this just checks for 3 failures in total

eleanor-merry commented on debian/chronos.init.d:
```diff
@@ -131,26 +131,31 @@ do_scale_operation()
     then
-      echo "SNMP statistics are not supported on this box. Please install the required packages and retry"
-      break
+      ((snmp_exists--))
```
This doesn't look right... make sure you test this

TygerBurning commented:
Updated
